pandas/core/missing.py

```diff
@@ -356,12 +369,21 @@ def _interpolate_scipy_wrapper(
             y = y.copy()
         if not new_x.flags.writeable:
             new_x = new_x.copy()
-        method = alt_methods[method]
-        new_y = method(x, y, new_x, **kwargs)
+
+        assert isinstance(method, str)
```

Comment:
same here: is there any guarantee we already checked this is a string by here?

Reply:
i'm not sure (don't think so) but if it's not then we'll raise with/without the assert because in the line below method is a key to a dict where all keys are strings
https://github.com/pandas-dev/pandas/blob/862cd05df4452592a99dd1a4fa10ce8cfb3766f7/pandas/core/missing.py#L308-L314
I can add a more informative error message here, though